File: app/src/main/java/com/besome/sketch/MainDrawer.java
package com.besome.sketch;

import static com.google.android.material.theme.overlay.MaterialThemeOverlay.wrap;

import android.app.Activity;
import android.content.Context;
import android.content.ContextWrapper;
import android.content.Intent;
import android.net.Uri;
import android.util.AttributeSet;

import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.core.view.WindowInsetsCompat;

import com.besome.sketch.help.ProgramInfoActivity;
import com.besome.sketch.tools.NewKeyStoreActivity;
import com.google.android.material.navigation.NavigationView;

import a.a.a.mB;
import dev.chrisbanes.insetter.Insetter;
import dev.chrisbanes.insetter.Side;
import mod.hilal.saif.activities.tools.AppSettings;
import pro.sketchware.R;
import pro.sketchware.activities.about.AboutActivity;

public class MainDrawer extends NavigationView {
    private static final int DEF_STYLE_RES = R.style.Widget_SketchwarePro_NavigationView_Main;

    public MainDrawer(@NonNull Context context) {
        this(context, null);
    }

    public MainDrawer(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, R.attr.navigationViewStyle);
    }

    public MainDrawer(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(wrap(context, attrs, defStyleAttr, DEF_STYLE_RES), attrs, defStyleAttr);
        context = getContext();

        var layoutDirection = context.getResources().getConfiguration().getLayoutDirection();
        Insetter.builder()
                .margin(WindowInsetsCompat.Type.navigationBars(),
                        Side.create(layoutDirection == LAYOUT_DIRECTION_LTR,
                                false, layoutDirection == LAYOUT_DIRECTION_RTL, false))
                .applyToView(this);

        inflateHeaderView(R.layout.main_drawer_header);
        inflateMenu(R.menu.main_drawer_menu);
        setNavigationItemSelectedListener(item -> {
            initializeSocialLinks(item.getItemId());
            initializeDrawerItems(item.getItemId());

            // Return false to prevent selection
            return false;
        });
    }

    private void initializeSocialLinks(@IdRes int id) {
        if (!mB.a()) {
            @StringRes int url = -1;
            if (id == R.id.social_discord) {
                url = R.string.link_discord_invite;
            } else if (id == R.id.social_telegram) {
                url = R.string.link_telegram_invite;
            } else if (id == R.id.social_github) {
                url = R.string.link_github_url;
            }

            if (url != -1) {
                openUrl(getContext().getString(url));
            }
        }
    }

    private void initializeDrawerItems(@IdRes int id) {
        Activity activity = unwrap(getContext());
        if (id == R.id.about_team) {
            Intent intent = new Intent(activity, AboutActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
            activity.startActivity(intent);
        } else if (id == R.id.changelog) {
            Intent intent = new Intent(activity, AboutActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
            intent.putExtra("select", "changelog");
            activity.startActivity(intent);
        } else if (id == R.id.program_info) {
            Intent intent = new Intent(activity, ProgramInfoActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
            activity.startActivityForResult(intent, 105);
        } else if (id == R.id.app_settings) {
            Intent intent = new Intent(activity, AppSettings.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
            activity.startActivity(intent);
        } else if (id == R.id.create_release_keystore) {
            Intent intent = new Intent(activity, NewKeyStoreActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
            activity.startActivity(intent);
        }
    }

    private void openUrl(String url) {
        Activity activity = unwrap(getContext());
        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
        activity.startActivity(intent);
    }

    private Activity unwrap(Context context) {
        while (!(context instanceof Activity) && context instanceof ContextWrapper) {
            context = ((ContextWrapper) context).getBaseContext();
        }

        return (Activity) context;
    }
}

File: app/src/main/java/com/besome/sketch/MainDrawer.kt
package com.besome.sketch

import a.a.a.mB
import android.app.Activity
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.util.AttributeSet
import androidx.annotation.IdRes
import androidx.core.view.WindowInsetsCompat
import com.besome.sketch.help.ProgramInfoActivity
import com.besome.sketch.tools.NewKeyStoreActivity
import com.google.android.material.navigation.NavigationView
import com.google.android.material.theme.overlay.MaterialThemeOverlay.wrap
import com.sketchware.remod.R
import dev.chrisbanes.insetter.Insetter
import dev.chrisbanes.insetter.Side
import mod.hilal.saif.activities.tools.AppSettings
import mod.ilyasse.activities.about.AboutActivity

class MainDrawer @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = R.attr.navigationViewStyle
) : NavigationView(
    wrap(context, attrs, defStyleAttr, R.style.Widget_SketchwarePro_NavigationView_Main),
    attrs,
    defStyleAttr
) {

    init {
        val layoutDirection = context.resources.configuration.layoutDirection
        Insetter.builder()
            .margin(
                WindowInsetsCompat.Type.navigationBars(),
                Side.create(
                    left = layoutDirection == LAYOUT_DIRECTION_LTR,
                    top = false,
                    right = layoutDirection == LAYOUT_DIRECTION_RTL,
                    bottom = false
                )
            )
            .applyToView(this)

        inflateHeaderView(R.layout.main_drawer_header)
        inflateMenu(R.menu.main_drawer_menu)
        setNavigationItemSelectedListener { item ->
            handleNavigationItemSelected(item.itemId)
            false // Prevent selection
        }
    }

    private fun handleNavigationItemSelected(@IdRes itemId: Int) {
        if (!mB.a()) {
            handleSocialLinks(itemId)
        }
        handleDrawerItems(itemId)
    }

    private fun handleSocialLinks(@IdRes itemId: Int) {
        val urlResId = when (itemId) {
            R.id.social_discord -> R.string.link_discord_invite
            R.id.social_telegram -> R.string.link_telegram_invite
            R.id.social_github -> R.string.link_github_url
            else -> null
        }

        urlResId?.let { openUrl(context.getString(it)) }
    }

    private fun handleDrawerItems(@IdRes itemId: Int) {
        when (itemId) {
            R.id.about_team -> startActivity<AboutActivity>()
            R.id.changelog -> startActivity<AboutActivity>("select" to "changelog")
            R.id.program_info -> startActivityForResult<ProgramInfoActivity>(105)
            R.id.app_settings -> startActivity<AppSettings>()
            R.id.create_release_keystore -> startActivity<NewKeyStoreActivity>()
        }
    }

    private inline fun <reified T : Activity> startActivity(vararg extras: Pair<String, String?>) {
        val intent = Intent(context, T::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
            extras.forEach { putExtra(it.first, it.second) }
        }
        context.startActivity(intent)
    }

    private inline fun <reified T : Activity> startActivityForResult(requestCode: Int) {
        val intent = Intent(context, T::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
        }
        (context as? Activity)?.startActivityForResult(intent, requestCode)
    }

    private fun openUrl(url: String) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        context.startActivity(intent)
    }
}